src/tls/verifier.rs

```diff
@@ -193,33 +192,25 @@ fn verify_presented_certs(presented_certs: &[Certificate]) -> Result<(), TLSErro
     certificate
         .check_self_issued()
         .map_err(TLSError::WebPKIError)?;
-    verify_libp2p_signature(&extension, certificate.subject_public_key_info().key())
+    verify_libp2p_signature(&extension, certificate.subject_public_key_info().spki())
         .map_err(TLSError::WebPKIError)
 }
 
-struct Libp2pExtension<'a> {
+struct Libp2pExtension {
     peer_key: PublicKey,
-    signature: &'a [u8],
+    signature: Vec<u8>,
```

Comment:
This is just another heap allocation without any clear justification I can see.

Reply:
To be honest, I did it to be compatible. I tested, it worked. I am fine if you try to somehow make it work without additional heap allocations. However I could not make it work =)